mbuild/compound.py

```diff
@@ -2236,6 +2349,200 @@ def _energy_minimize_openbabel(
+                                "which dimensions to fix motion. "
+                            )
+                    # if the list is defined as [[compound1],[compound2]]
+                    elif len(fixed_temp) == 1:
+                        p1 = fixed_temp[0]
+                        dims = [True, True, True]
+                else:
+                    p1 = fixed_temp
+                    dims = [True, True, True]
+
+                all_true = False
+                if dims[0] == dims[1] == dims[2] == True:
+                    all_true = True
+
+                if id(p1) != id(self):
+                    if id(p1) not in list(
+                        map(lambda x: id(x), self.successors())
+                    ):
+                        raise MBuildError(
+                            f"{p1} is not a member of Compound {self}."
+                        )
+
+                if len(p1.children) == 0:
+                    index = 1
+                    for particle in self.particles():
+                        if id(p1) == id(particle):
+                            pid = index
+                            if all_true == True:
+                                ob_constraints.AddAtomConstraint(pid)
+                            else:
+                                if dims[0] == True:
+                                    ob_constraints.AddAtomXConstraint(pid)
+                                if dims[1] == True:
+                                    ob_constraints.AddAtomYConstraint(pid)
+                                if dims[2] == True:
+                                    ob_constraints.AddAtomZConstraint(pid)
+                        index += 1
+                else:
+                    for successor in p1.successors():
+                        index = 1
+                        for particle in self.particles():
+                            if id(successor) == id(particle):
+                                pid = index
+                                if all_true == True:
+                                    ob_constraints.AddAtomConstraint(pid)
+                                elif dims[0] == True:
+                                    ob_constraints.AddAtomXConstraint(pid)
+                                elif dims[1] == True:
+                                    ob_constraints.AddAtomYConstraint(pid)
+                                elif dims[2] == True:
+                                    ob_constraints.AddAtomZConstraint(pid)
+                            index += 1
+
+        if ignore_compounds is not None:
+            temp1 = np.array(ignore_compounds)
+            if len(temp1.shape) == 2:
+                ignore_compounds = list(temp1.reshape(-1))
+            for ignore in ignore_compounds:
+                p1 = ignore
+                if id(p1) != id(self):
+                    if id(p1) not in list(
+                        map(lambda x: id(x), self.successors())
+                    ):
```

Comment:
```suggestion
                if id(p1) != id(self):
                    if id(p1) not in particle_idx:
```
Refer to `particle_idx` dict from above